### ISSUE TITLE
Scene plugins cannot receive configuration data

### ISSUE THREAD

**Issue by TJ09** [photonstorm/phaser#4471]
## Version
* Phaser Version: 3.16.2

## Description

ScenePlugin inherits the [documentation of the `init()` method](https://photonstorm.github.io/phaser3-docs/Phaser.Plugins.ScenePlugin.html#init__anchor) from BasePlugin, which says that init will be called with the data passed to the plugin config.

Similarly, the [`PluginObject`](https://photonstorm.github.io/phaser3-docs/global.html#PluginObject) type says that the scene config is an array of [`PluginObjectItem`](https://photonstorm.github.io/phaser3-docs/global.html#PluginObject), which has a field called `data` that is "*[a]rbitrary data passed to the plugin's init() method*."

Unfortunately, neither of these are true. PluginManager constructs scene plugins but never actually passes data to them:
https://github.com/photonstorm/phaser/blob/974ceb5654edb9db552b1343fc2bb3f7f69d464d/src/plugins/PluginManager.js#L257-L275
unlike the global case where `init` is indeed called:
https://github.com/photonstorm/phaser/blob/974ceb5654edb9db552b1343fc2bb3f7f69d464d/src/plugins/PluginManager.js#L576-L589

## Example Test Code

```js
var TestBasePlugin = new Phaser.Class({
    Extends: Phaser.Plugins.BasePlugin,

    init: function(data) {
        console.log('TestBasePlugin initialized with', data);
    }
});

var TestScenePlugin = new Phaser.Class({
    Extends: Phaser.Plugins.ScenePlugin,

    init: function(data) {
        console.log('TestScenePlugin initialized with', data);
    }
});

var game = new Phaser.Game({
    type: Phaser.HEADLESS,
    parent: 'phaser-example',
    plugins: {
        global: [
            { key: 'TestBasePlugin', plugin: TestBasePlugin, start: true, data: "test" },
        ],
        scene: [
            { key: 'TestScenePlugin', plugin: TestScenePlugin, mapping: 'testScenePlugin', start: true, data: "test" },
        ],
    },
    scene: {
        create: function() {
            console.log('Scene created', this.testScenePlugin);
        },
    }
});
```

### Expected console output
```
TestBasePlugin initialized with test
TestScenePlugin initialized with test
Scene created ...
```

### Actual console output
```
TestBasePlugin initialized with test
Scene created ...
```

**Comment by samme** [comment]
Technically ScenePlugin extends BasePlugin but functionally they’re quite different. So the docs are ambiguous, unfortunately.

I believe only global plugins (BasePlugin) use `init()`, `start()`, and `stop()`, and only scene plugins use `boot()`.

**Comment by TJ09** [comment]
> I believe only global plugins (BasePlugin) use init(), start(), and stop(), and only scene plugins use boot().

Yes, though it would be good for the documentation to reflect this (e.g. by making game-level plugins a separate class that also extends BasePlugin, and updating the documentation for PluginObject/PluginObjectItem).

Though it'd also be nice for the data passed to a plugin config to be accessible *somewhere*, even if not via `init`.

**Comment by samme** [comment]
If you’re writing your own plugins, you can do this ad hoc by reading your chosen property in `scene.sys.settings.data` when the plugin is instantiated or booted.

**Comment by samme** [comment]
I think that [BasePlugin#boot](https://photonstorm.github.io/phaser3-docs/Phaser.Plugins.BasePlugin.html#boot__anchor), [BasePlugin#scene](https://photonstorm.github.io/phaser3-docs/Phaser.Plugins.BasePlugin.html#scene__anchor), and [BasePlugin#systems](https://photonstorm.github.io/phaser3-docs/Phaser.Plugins.BasePlugin.html#systems__anchor) could all be removed.

**Comment by TJ09** [comment]
`scene.sys.settings.data` represents the data passed to a specific scene, rather than the data passed to a specific plugin.

I've been able to access the `PluginObject` config via something like:
```js
const data = scene.game.config.installScenePlugins
  .filter(pluginObj => pluginObj.plugin === this.constructor)[0]
  .data;
```

Which is messy but at least it works.

**Comment by photonstorm** [comment]
Closing as this was solved in the merge of #4488 a long time ago!
